grammar/parsing.js

```diff
@@ -206,12 +206,80 @@ class Parsing extends EmbeddedActionsParser {
             return result;
         });
 
+        $.RULE('multiArray', () => {
+            const multiArr = [[]];
+            let singleArr = null;
+
+            const updateRow = (multiArr, newArr) => {
+                if (multiArr[0] === null)
+                    multiArr.push([])
+                for(let newArrIndex = 0; newArrIndex < newArr.length; newArrIndex++) {
+                    let currVal = newArr[newArrIndex];
+                    if (!Array.isArray(currVal)) {
+                        multiArr[0].push(currVal);
+                        continue;
```

Comment:
Note: inside an `Array.map()`, you cannot use `continue`; but you can use `return` to short-circuit out of a given loop.